go/common/identity/identity.go

```diff
@@ -86,40 +105,22 @@ func doLoadOrGenerate(dataDir string, signerFactory signature.SignerFactory, sho
 		signers = append(signers, signer)
 	}
 
-	// TLS certificate.
-	//
-	// TODO: The key and cert could probably be made totally ephemeral, as long
-	// as the registry update takes effect immediately.
-	var (
-		cert *tls.Certificate
-		err  error
-	)
-	tlsCertPath, tlsKeyPath := TLSCertPaths(dataDir)
-	if shouldGenerate {
-		cert, err = tlsCert.LoadOrGenerate(tlsCertPath, tlsKeyPath, CommonName)
-	} else {
-		cert, err = tlsCert.Load(tlsCertPath, tlsKeyPath)
+	// TLS certificate is always freshly generated.
+	cert, err := tlsCert.Generate(CommonName)
+	if err != nil {
+		return nil, err
 	}
+	nextCert, err := tlsCert.Generate(CommonName)
```

Comment:
Also make sure to reduce certificate expiration once this works.

Reply:
Sentry nodes and IAS proxies require non-rotating certificates, so I've left this as-is for now.